pkg/forklift-api/webhooks/mutating-webhook/mutators/plan-mutator.go

```diff
@@ -0,0 +1,144 @@
+package mutators
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	net "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/konveyor/forklift-controller/pkg/apis"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	core "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PlanMutator struct {
+}
+
+func (mutator *PlanMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("plan mutator was called")
+	raw := ar.Request.Object.Raw
+	plan := &api.Plan{}
+	err := json.Unmarshal(raw, plan)
+	if err != nil {
+		log.Error(err, "mutating webhook error, failed to unmarshel plan")
+		return util.ToAdmissionResponseError(err)
+	}
+
+	var planChanged bool
+
+	if plan.Spec.TransferNetwork == nil {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			log.Error(err, "Couldn't get the cluster configuration", err.Error())
+			return util.ToAdmissionResponseError(err)
+		}
+
+		err = api.SchemeBuilder.AddToScheme(scheme.Scheme)
+		if err != nil {
+			log.Error(err, "Couldn't build the scheme", err.Error())
+			return util.ToAdmissionResponseError(err)
+		}
+		err = apis.AddToScheme(scheme.Scheme)
+		if err != nil {
+			log.Error(err, "Couldn't add forklift API to the scheme", err.Error())
+			return util.ToAdmissionResponseError(err)
+		}
+		err = net.AddToScheme(scheme.Scheme)
+		if err != nil {
+			log.Error(err, "Couldn't add network-attachment-definition-client to the scheme", err.Error())
+			return util.ToAdmissionResponseError(err)
+		}
+
+		cl, err := client.New(config, client.Options{Scheme: scheme.Scheme})
+		if err != nil {
+			log.Error(err, "Couldn't create a cluster client", err.Error())
+			return util.ToAdmissionResponseError(err)
+		}
+
+		targetProvider := api.Provider{}
+		err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Destination.Namespace, Name: plan.Spec.Provider.Destination.Name}, &targetProvider)
+		if err != nil {
+			log.Error(err, "Couldn't get the target provider", err.Error())
+			return util.ToAdmissionResponseError(err)
+		}
+
+		if network := targetProvider.Annotations["forklift.konveyor.io/defaultTransferNetwork"]; network != "" {
+			key := client.ObjectKey{
+				Namespace: plan.Spec.TargetNamespace,
+				Name:      network,
+			}
+
+			var rcl client.Client // remote client, i.e., client to a possibly remote cluster
```

Comment:
can't we use the same client for both local and remote?

Reply:
I've posted an optimization that uses the existing client to local cluster in case the local cluster is the target but it seems you meant something else here - what did you mean to use the same client for both? the URL is different, the credentials may be different..

Reply:
i meant why using `cl` instead of initiating the `tcl` from the beginning.
```
var tcl client.Client // target client, i.e., client to a possibly remote cluster
			if targetProvider.IsHost() {
				tcl = cl
			} else {
				ref := targetProvider.Spec.Secret
```
doing it above and dropping the `cl`.

Reply:
ah that's because not all entities are available on the target namespace, the mutator needs the provider that exists only on the local cluster